test/test_tcp_sender.py

```diff
@@ -0,0 +1,59 @@
+import queue
+from unittest import mock
+import ros_tcp_endpoint
+
+
+@mock.patch("ros_tcp_endpoint.tcp_sender.rospy")
+def test_tcp_sender_constructor(mock_ros):
+    tcp_sender = ros_tcp_endpoint.tcp_sender.UnityTcpSender()
+    assert tcp_sender.sender_id == 1
+    assert tcp_sender.time_between_halt_checks == 5
+    assert tcp_sender.queue == None
+    assert tcp_sender.next_srv_id == 1001
+    assert tcp_sender.srv_lock != None
+    assert tcp_sender.services_waiting == {}
+
+
+@mock.patch.object(ros_tcp_endpoint.tcp_sender.UnityTcpSender, "send_unity_message")
+def test_send_unity_error_should_send_msg(mock_send_message):
+    sender = ros_tcp_endpoint.tcp_sender.UnityTcpSender()
+    sender.send_unity_error("Test error")
+    mock_send_message.assert_called_once()
+
+
+@mock.patch.object(ros_tcp_endpoint.client.ClientThread, "serialize_message")
+def test_send_message_should_serialize_message(mock_serialize_msg):
+    sender = ros_tcp_endpoint.tcp_sender.UnityTcpSender()
+    sender.send_unity_message("test topic", "test msg")
```

Comment:
Is it possible to assert the serialized bytes too?

Reply:
We should be able to test the serialized bytes. I will look into it.